src/core/algorithms/od/fastod/partitions/complex_stripped_partition.cpp

```diff
@@ -93,6 +93,115 @@ void ComplexStrippedPartition::ToStrippedPartition() {
     should_be_converted_to_sp_ = false;
 }
 
+template <bool Ascending>
+bool ComplexStrippedPartition::Swap(model::ColumnIndex left, model::ColumnIndex right) const {
+    size_t const group_count = is_stripped_partition_ ? sp_begins_->size() : rb_begins_->size();
+
+    for (size_t begin_pointer = 0; begin_pointer < group_count - 1; begin_pointer++) {
+        size_t const group_begin = is_stripped_partition_ ? (*sp_begins_)[begin_pointer]
+                                                          : (*rb_begins_)[begin_pointer];
+
+        size_t const group_end = is_stripped_partition_ ? (*sp_begins_)[begin_pointer + 1]
+                                                        : (*rb_begins_)[begin_pointer + 1];
+
+        std::vector<std::pair<int, int>> values;
+
+        if (is_stripped_partition_) {
+            values.reserve(group_end - group_begin);
+
+            for (size_t i = group_begin; i < group_end; ++i) {
+                size_t const index = (*sp_indexes_)[i];
+
+                values.emplace_back(data_->GetValue(index, left), data_->GetValue(index, right));
+            }
+        } else {
+            for (size_t i = group_begin; i < group_end; ++i) {
+                DataFrame::Range const range = (*rb_indexes_)[i];
+
+                for (size_t j = range.first; j <= range.second; ++j) {
+                    values.emplace_back(data_->GetValue(j, left), data_->GetValue(j, right));
+                }
+            }
+        }
```

Comment:
IDK if I should comment on this code since you've just moved it to another place, but the `else` branch does not have the `values.reserve`